src/fsharp/pars.fsy

```diff
@@ -962,13 +962,15 @@ classMemberSpfn:
        SynMemberSig.Inherit ($4,unionRanges (rhs parseState 3) ($4).Range) }
 
   | opt_attributes opt_declVisibility VAL fieldDecl 
-     { if Option.isSome $2 then errorR(Error(FSComp.SR.parsVisibilityDeclarationsShouldComePriorToIdentifier(),rhs parseState 2))
-       let fld = $4 $1 false 
-       SynMemberSig.ValField(fld,rhs2 parseState 3 4) }
+     { let wholeRange = rhs2 parseState 1 4
```

Comment:
Is there a reason for computing wholeRange prior to the errorcheck?

Reply:
Since we're doing a recovery for the error it doesn't really matter which order is used. I'll change it if you really want it but I find it just a matter of style.

Reply:
If the work is not necessary for evaluating the error, then it probably makes sense to move it. :-)

Reply:
I notice the last set of changes, also had this pattern.  So, I guess there is precedent in the codebase :-)

Reply:
Yea, there is already precedent, so it's fair to keep it as is.